lib/config.c

```diff
@@ -428,14 +429,53 @@ walk_host_config(
                 log_internal(MTC_LOG_ERR, "%s: failed to get IPaddress\n", __func__);
                 return MTC_ERROR_CF_INVALID_FORMAT;
             }
-            c->common.host[c->common.hostnum].ip_address = inet_addr((char *) txt);
-            if (c->common.host[c->common.hostnum].ip_address == 0) 
+
+            const char *ip_address = (char *) txt;
+
+            struct addrinfo hints;
+            struct addrinfo *result;
+            memset(&hints, 0, sizeof hints);
+            hints.ai_family = AF_UNSPEC;
+            hints.ai_socktype = SOCK_DGRAM;
+            hints.ai_flags = 0;
```

Comment:
You might want to avoid doing a DNS lookup here and just parse the numeric IP, there is a flag for that: AI_NUMERICHOST.
(DNS may or may not work yet and may introduce a long pause to get a timeout).